### PR TITLE
initial nrf52_upload extraction from pc-nrfutil

### LINKED ISSUE #265: pynitrokey >=0.4.26 not installable on Arm
On Arm systems, pip only installs pynitrokey up to version 0.4.25.  This is probably due to the indirect dependency `pc_ble_driver_py` (pulled in by `nrfutil`), see https://github.com/NordicSemiconductor/pc-ble-driver-py/issues/202.

Related issues:
- https://github.com/NordicSemiconductor/pc-nrfutil/issues/337
- https://github.com/NordicSemiconductor/pc-nrfutil/issues/361
- https://github.com/NordicSemiconductor/pc-ble-driver/pull/271

### LINKED ISSUE #330: Python 3.11 Support
The issue is currently being tracked at https://github.com/NordicSemiconductor/pc-ble-driver-py/issues/230

In the meantime, is there a way to circumvent this dependency for [pynitrokey](https://github.com/Nitrokey/pynitrokey)? Like an option to self-build the [pc-ble-driver-py](https://github.com/NordicSemiconductor/pc-ble-driver-py) component when installing through `pipx`?

### PR DESCRIPTION
**Pull request by daringer**
This PR replaces the `nrfutil` dependency with an extract from the originating, archived repository.

## Changes
* use `nk3.bootloader.nrf52_upload` instead of `nrfutil` to upload a signed firmware to a Nitrokey 3 Mini (nRF52)
* avoids importing `pc-ble-driver-py`
* enables python 3.11 support
* (untested) should enable arm / *bsd support
* adds `pyserial` & `protobuf` to dependencies
* comes with the necessary `package.py` to create firmware images, although it is not used
* applies `black` & `isort` to `nrf52_upload`

## Checklist

- [x] tested with Python3.9
- [x] run `make check` or `make fix` for the formatting check
- [x] signed commits
- [x] updated documentation (e.g. parameter description, inline doc, docs.nitrokey)

## Test Environment and Execution

- OS: Arch
- device's model: nk3am
- device's firmware version: 1.3.0RC1

Fixes #330 
Fixes #265

### CODE REVIEW

**Comment by daringer** [comment]
From technical point of view I would say this is fine for an initial working state.
I added a README.md, maybe you could take a look, if this is ok from license point of view.

* The sources are not in their original state anymore due to `black` and `isort`, don't see an issue with this
* How about removing the license notices from each file and replacing them with spdx + the complete license within the parent directory ? 
* any other license related things to consider?

File: pynitrokey/nk3/bootloader/__init__.py [hunk 1]
```diff
@@ -127,10 +127,10 @@ def validate_firmware_image(
...
 
     if not metadata.signed_by_nitrokey:
-        raise Exception(
-            f"Firmware image is not signed by Nitrokey (signed by: {metadata.signed_by})"
-        )
-
+        # raise Exception(
+        #    f"Firmware image is not signed by Nitrokey (signed by: {metadata.signed_by})"
+        # )
+        print("not an officially signed image!")
```
Comment (robin-nitrokey) [inline pynitrokey/nk3/bootloader/__init__.py:133]: revert?

**Review by robin-nitrokey — APPROVED** [review]
As we discussed, I’d prefer to have this in a separate package to have a clean separation, otherwise looks good to me.

File: pynitrokey/nk3/bootloader/__init__.py (hunk 1 above)
Comment (daringer) [reply pynitrokey/nk3/bootloader/__init__.py]: oh yes! done

**Comment by daringer** [comment]
windows tests still missing before we can merge ...

... separate package:  `Nitrokey/nrf52-uploader` ?

**Pull request merged**